js/views/modals/Settings/Store.js

```diff
@@ -0,0 +1,325 @@
+    });
+
+    this.modsAvailable = new Moderators(null, {
+      async: true,
+      include: 'profile',
+      excludeCollection: this.modsSelected,
+    });
+
+    this.listenTo(this.modsSelected, 'add', (model, collection) => {
+      this.buildModList(model, collection, this.$modListSelected, { cardState: 'selected' });
+    });
+
+    this.listenTo(this.modsByID, 'add', (model, collection) => {
+      this.buildModList(model, collection, this.$modListByID);
+    });
+
+    this.listenTo(this.modsAvailable, 'add', (model, collection) => {
+      this.buildModList(model, collection, this.$modListAvailable);
+    });
+
+    this.listenTo(this.profile, 'sync', () => app.profile.set(this.profile.toJSON()));
+    this.listenTo(this.settings, 'sync', () => app.settings.set(this.settings.toJSON()));
+  }
+
+  events() {
+    return {
+      'click .js-browseMods': 'fetchAvailableModerators',
+      'click .js-submitModByID': 'clickSubmitModByID',
+      'click .js-save': 'save',
+    };
+  }
+
+  fetchAvailableModerators() {
+    // be aware that this call can take a long time
+    this.$modListAvailable.addClass('processing');
+    this.modsAvailable.fetch()
+      .fail((...args) => {
+        const title = app.polyglot.t('settings.storeTab.errors.availableModsFailed');
+        const message = args[0] && args[0].responseJSON && args[0].responseJSON.reason || '';
+        openSimpleMessage(title, message);
+      })
+      .always(() => {
+        setTimeout(() => {
+          // remove the processing class after a long enough time. If it's still visible
+          // there are probably no moderators coming.
+          this.$modListAvailable.removeClass('processing');
+        }, 15000);
+      });
+  }
+
+  buildModList(model, collection, target, opts = {}) {
+    target.toggleClass('hasMods', !!collection.length);
+
+    if (model) {
+      const docFrag = $(document.createDocumentFragment());
+      const newMod = this.createChild(ModCard, {
+        model,
+        ...opts,
+      });
+      this.listenTo(newMod, 'changeModerator', (data) => this.changeMod(data));
+      docFrag.append(newMod.render().$el);
+      target.append(docFrag);
+    }
+
+    // if all moderators have loaded, clear any processing class
+    if (!collection.notFetchedYet.length) {
+      target.removeClass('processing');
+    }
+  }
+
+  clickSubmitModByID() {
+    const modID = this.$submitModByIDInput.val();
+    const blankError = app.polyglot.t('settings.storeTab.errors.modIsBlank');
+
+    this.$submitModByIDInputError.addClass('hide');
+
+    if (modID) {
+      this.$submitModByID.addClass('processing');
+      this.processIDorHandle(modID);
+    } else {
+      this.$submitModByIDInputError.removeClass('hide');
+      this.$submitModByIDInputErrorText.text(blankError);
+    }
+  }
+
+  processIDorHandle(modID) {
+    if (modID.charAt(0) === '@') {
+      // if the id is a handle, get the guid
+      const handle = modID.slice(1);
+      getGuid(handle)
```

Comment:
There needs to be some cleanup on any async calls like this one (or `this.modsSelected.fetch({ fetchList: this.currentMods });`).

1.) In remove of this view, the calls should be aborted.
2.) In the handlers, the first line should be something to the effect of `if (this.isRemoved) return;`).

Otherwise, you'll have the situation where the user navigates away from this view and the handlers of those calls still execute down the line. At the very least, their keeping this view from being garbage collected. On a worse level, they may trigger Dialogs that are no longer applicable.